src/pybamm/util.py

```diff
@@ -23,6 +23,40 @@ def root_dir():
 
 
 class FuzzyDict(dict):
+    """
+    A dictionary with fuzzy matching capabilities for key retrieval and search.
+
+    Methods
+    -------
+    get_best_matches(key)
+        Returns a list of the best-matching keys for a given input key.
+
+    search(keys, print_values=False)
+        Searches the dictionary for keys containing all specified terms. Prints
+        the results and, optionally, the corresponding values.
+
+    copy()
+        Returns a copy of the FuzzyDict instance.
+
+    Example usage
+    -------
+    ```python
+        >>> params = FuzzyDict({
+        >>> "electrode diffusivity": 1.2,
+        >>> "particle diffusivity": 0.8,
+        >>> "Negative electrode SOC": 0.5,
+        >>> "Open-circuit voltage at 100% SOC [V]": 4.2,
+        >>> })
+
+        >>> print(params["electrode diffusivity"])  # 1.2 (direct access)
+        >>> print(params["particle diffusivity"])   # 0.8 (handles renaming)
+
+        >>> params.search("open circuit voltage")
+
+        >>> params_copy = params.copy()  # Creates a new FuzzyDict instance
+    ```
+    """
```

Comment:
You will need to fix this examples as the doctests will fail
```suggestion
    Examples
    --------
      >>> params = FuzzyDict({
      >>> "electrode diffusivity": 1.2,
      >>> "particle diffusivity": 0.8,
      >>> "Negative electrode SOC": 0.5,
      >>> "Open-circuit voltage at 100% SOC [V]": 4.2,
      >>> })
      >>> print(params["electrode diffusivity"])
      1.2
      >>> print(params["particle diffusivity"])
      0.8
      >>> params.search("open circuit voltage")
      >>> params_copy = params.copy()  # Creates a new FuzzyDict instance
    """
```

Reply:
Modified the example.